tests/testing_common.py

```diff
@@ -870,70 +873,95 @@ def _test_weighted_combination_of_adapters(self, model_id, config_cls, config_kw
-                        self.assertTrue(
-                            torch.allclose(new_delta_weight, weighted_original_delta_weights, atol=1e-4, rtol=1e-4)
-                        )
-                    elif "svd" in adapter_name:
-                        self.assertTrue(target.r[adapter_name] == 20)
-                    elif "linear" in adapter_name:
-                        self.assertTrue(target.r[adapter_name] == 8)
-                    elif "cat" in adapter_name:
-                        self.assertTrue(target.r[adapter_name] == 28)
-
-        for adapter_name in new_adapters:
-            # ensuring new adapters pass the forward loop
-            model.set_adapter(adapter_name)
-            dummy_input = self.prepare_inputs_for_testing()
-            model.eval()
-            _ = model(**dummy_input)[0]
+            with self.assertRaises(ValueError):
+                model.add_weighted_adapter(
+                    adapter_list[1:],
+                    weight_list[1:],
+                    "multi_adapter_linear_reweighting_uneven_r",
+                    combination_type="linear",
+                )
+
+            new_adapters = [
+                "single_adapter_reweighting",
+                "multi_adapter_svd_reweighting",
+                "multi_adapter_cat_reweighting",
+                "multi_adapter_linear_reweighting",
+            ]
+            for new_adapter in new_adapters:
+                self.assertTrue(new_adapter in model.peft_config)
+
+            key_list = [key for key, _ in model.named_modules() if "lora" not in key]
+            for key in key_list:
+                _, target, _ = _get_submodules(model, key)
+                if isinstance(target, LoraLayer):
+                    for adapter_name in new_adapters:
+                        if "single" in adapter_name:
+                            new_delta_weight = target.get_delta_weight(adapter_name)
+                            weighted_original_delta_weights = target.get_delta_weight(adapter_list[0]) * weight_list[0]
+                            self.assertTrue(
+                                torch.allclose(new_delta_weight, weighted_original_delta_weights, atol=1e-4, rtol=1e-4)
+                            )
+                        elif "svd" in adapter_name:
+                            self.assertTrue(target.r[adapter_name] == 20)
+                        elif "linear" in adapter_name:
+                            self.assertTrue(target.r[adapter_name] == 8)
+                        elif "cat" in adapter_name:
+                            self.assertTrue(target.r[adapter_name] == 28)
+
+            for adapter_name in new_adapters:
+                # ensuring new adapters pass the forward loop
+                model.set_adapter(adapter_name)
+                dummy_input = self.prepare_inputs_for_testing()
+                model.eval()
+                _ = model(**dummy_input)[0]
+
+        elif isinstance(config, (IA3Config)):
```

Comment:
`Linear` for $(IA)^3$ does not have a `get_delta_weight` method. How should we test that the result is correct?